packages/firestore/lite/src/api/database.ts

```diff
@@ -63,12 +75,42 @@ export class Firestore implements firestore.FirebaseFirestore {
       );
     }
     this._settings = settings;
+
+    const databaseInfo = this._makeDatabaseInfo();
+
+    // Kick off initializing the datastore but don't actually wait for it.
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    PlatformSupport.getPlatform()
+      .loadConnection(databaseInfo)
+      .then(connection => {
+        const databaseInfo = this._makeDatabaseInfo();
+        const serializer = PlatformSupport.getPlatform().newSerializer(
+          databaseInfo.databaseId
+        );
+        this._datastore = newDatastore(
+          connection,
+          this._credentials,
+          serializer
+        );
+        this._initializationDone.resolve(this);
+      });
   }
 
-  _ensureClientConfigured(): void {
+  _ensureClientConfigured(): Promise<Firestore> {
     if (!this._settings) {
       this._settings = {};
     }
+    return this._initializationDone.promise;
+  }
+
+  private _makeDatabaseInfo(): DatabaseInfo {
```

Comment:
Similar to https://osscs.corp.google.com/firebase-sdk/firebase-js-sdk/+/master:packages/firestore/src/api/database.ts;l=494?q=makeDatabaseInfo&ss=firebase-sdk%2Ffirebase-js-sdk